src/service/api.js

```diff
@@ -17,7 +18,20 @@ class Service {
     this.service = service;
   }
 
-  handleRequestSuccess(config) {
+  async handleRequestSuccess(config) {
+    if (!store.state.auth.tokens) {
+      return config;
+    }
+
+    const { access } = store.state.auth.tokens;
+    if (access) {
+      // Will trigger infinite loop if trying to refresh on refresh call
```

Comment:
Just a suggestion. Instead of checking for `startsWith("token/")`, you could simply setup something like `plainService` on the `API` class, use that for things unauthenticated calls (like setting up tokens in your case). Would avoid the loop.

Otherwise good to merge, your call.

Reply:
I think right now it is ok to simply merge, but this change or maybe adding a `whitelist` property to the service would make sense indeed